### PR TITLE
Add `traceOptionsRequests` option to disable tracing of OPTIONS requests

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

`traceOptionsRequests` defaults to `true` for now. I think we should change this to `false` only in the next major release.

## :bulb: Motivation and Context

Allows us to fix https://github.com/getsentry/sentry-java/issues/2231 in the next major by flipping the default to `false`.

## :green_heart: How did you test it?
Unit Tests, manually

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/SentryOptions.java [hunk 1]
```diff
@@ -387,6 +387,10 @@ public class SentryOptions {
 
   private @NotNull IMemoryCollector memoryCollector = NoOpMemoryCollector.getInstance();
 
+  // TODO this should default to false on the next major
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/SentryOptions.java:390]: Do you agree with this approach of having it default to `true` until the next major then change it to the desired default of `false`?

**Comment by adinauer** [comment]
Docs PR here: https://github.com/getsentry/sentry-docs/pull/6019

File: sentry/src/main/java/io/sentry/SentryOptions.java (hunk 1 above)
Comment (markushi) [reply sentry/src/main/java/io/sentry/SentryOptions.java:390]: Sounds good 👍

**Review by markushi — APPROVED** [review]
LGTM!

**Review by romtsn — APPROVED** [review]

**Pull request merged**